### PR TITLE
Modify `govet` settings

### PR DESCRIPTION
**Pull request by dhrubabasu**
## Why this should be merged

Catches some more stuff

## How this works

pretty straightforward

## How this was tested

CI

### CODE REVIEW

**Review by JacobEverly — APPROVED** [review]

File: snow/networking/benchlist/benchlist_test.go [hunk 1]
```diff
@@ -218,9 +218,6 @@ func TestBenchlistMaxStake(t *testing.T) {
 	require.Equal(b.benchedQueue.Len(), 2)
 	require.Equal(b.benchlistSet.Len(), 2)
 	require.Len(b.failureStreaks, 1)
-	fs := b.failureStreaks[vdrID2]
-	fs.consecutive = threshold
-	fs.firstFailure = now
```
Comment (abi87) [inline snow/networking/benchlist/benchlist_test.go:223]: why dropping this?
Why not
```
b.failureStreaks[vdrID2].consecutive = threshold
b.failureStreaks[vdrID2].firstFailure = now
```

Comment (dhrubabasu) [reply snow/networking/benchlist/benchlist_test.go:223]: That's not valid golang- `b.failureStreaks[vdrID2]` is not addressable

File: snow/engine/snowman/transitive_test.go [hunk 2]
```diff
@@ -1190,9 +1190,6 @@ func TestEngineRetryFetch(t *testing.T) {
 			IDV:     ids.GenerateTestID(),
 			StatusV: choices.Unknown,
 		},
-		ParentV: gBlk.ID(),
-		HeightV: 1,
-		BytesV:  []byte{1},
```
Comment (abi87) [inline snow/engine/snowman/transitive_test.go:1195]: So I understand these fields are not really used, but I feel they provide context that make the test more readable? They seems to tell intentions of test creator. I'd like to keep them, but open to discussion

File: snow/networking/benchlist/benchlist_test.go (hunk 1 above)
Comment (abi87) [reply snow/networking/benchlist/benchlist_test.go:223]: Right. So how about
```
	b.failureStreaks[vdrID2] = failureStreak{
		consecutive:  threshold,
		firstFailure: now,
	}
```

What I am saying is that maybe test creator intent was to set the failureStreak specific values.

**Comment by dhrubabasu** [comment]
Closing for now, will revisit in the future

**Pull request closed without merging**